packages/web3-shared/evm/utils/transaction.ts

```diff
@@ -9,10 +9,18 @@ export function isFinalState(type: TransactionStateType) {
     return [TransactionStateType.CONFIRMED, TransactionStateType.FAILED].includes(type)
 }
 
+/**
+ * FAILED -> UNKNOWN
```

Comment:
`FAILED -> UNKNOWN`  add this transition path.

Reply:
I draw a diagram for it with mermaid

[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVEQ7XG4gICBVTktOT1dOLS0-V0FJVF9GT1JfQ09ORklSTUlORztcbiAgIFVOS05PV04tLT5IQVNIO1xuICAgVU5LTk9XTi0tPlJFQ0VJUFQ7XG4gICBVTktOT1dOLS0-Q09ORklSTUVEO1xuICAgVU5LTk9XTi0tPkZBSUxFRDtcbiAgIFdBSVRfRk9SX0NPTkZJUk1JTkctLT5IQVNIO1xuICAgV0FJVF9GT1JfQ09ORklSTUlORy0tPlJFQ0VJUFQ7XG4gICBXQUlUX0ZPUl9DT05GSVJNSU5HLS0-Q09ORklSTUVEO1xuICAgV0FJVF9GT1JfQ09ORklSTUlORy0tPkZBSUxFRDtcbiAgIEhBU0gtLT5SRUNFSVBUO1xuICAgSEFTSC0tPkNPTkZJUk1FRDtcbiAgIEhBU0gtLT5GQUlMRUQ7XG4gICBSRUNFSVBULS0-Q09ORklSTUVEO1xuICAgUkVDRUlQVC0tPkZBSUxFRDtcbiAgICAgIEZBSUxFRC0tPlVOS05PV047XG5cbiIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2UsImF1dG9TeW5jIjp0cnVlLCJ1cGRhdGVEaWFncmFtIjpmYWxzZX0)](https://mermaid.live/edit#eyJjb2RlIjoiZ3JhcGggVEQ7XG4gICBVTktOT1dOLS0-V0FJVF9GT1JfQ09ORklSTUlORztcbiAgIFVOS05PV04tLT5IQVNIO1xuICAgVU5LTk9XTi0tPlJFQ0VJUFQ7XG4gICBVTktOT1dOLS0-Q09ORklSTUVEO1xuICAgVU5LTk9XTi0tPkZBSUxFRDtcbiAgIFdBSVRfRk9SX0NPTkZJUk1JTkctLT5IQVNIO1xuICAgV0FJVF9GT1JfQ09ORklSTUlORy0tPlJFQ0VJUFQ7XG4gICBXQUlUX0ZPUl9DT05GSVJNSU5HLS0-Q09ORklSTUVEO1xuICAgV0FJVF9GT1JfQ09ORklSTUlORy0tPkZBSUxFRDtcbiAgIEhBU0gtLT5SRUNFSVBUO1xuICAgSEFTSC0tPkNPTkZJUk1FRDtcbiAgIEhBU0gtLT5GQUlMRUQ7XG4gICBSRUNFSVBULS0-Q09ORklSTUVEO1xuICAgUkVDRUlQVC0tPkZBSUxFRDtcbiAgICAgIEZBSUxFRC0tPlVOS05PV047XG5cbiIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)

Reply:
It should not allow a `FAILED` tx to reset to `UNKNOWN`. Otherwise, should let `CONFIRMED` tx do that too. Why do we need it?

Reply:
As we had discussed, the state are shared by multiple transactions (not parallel), so it could be reset from `FAILED` to `UNKNOWN`